### PR TITLE
Ac404 72 implement adding merchants

### PR DESCRIPTION
**Pull request by mkljaic20**
Adding merchants is fully connected to the backend, it now sends card types and status in correct format. Also replaced hardcoded JWT to function call so it works with the logged in user.

### CODE REVIEW

File: app/src/main/java/com/Found404/paypro/ui/pages/CardPaymentsPage.kt [hunk 1]
```diff
@@ -93,22 +105,21 @@ fun CardPayments(
...
-                ) {
-                    merchantModel = if (merchantModel.cardTypes.contains(cardType.name)) {
-                        merchantModel.copy(cardTypes = merchantModel.cardTypes - cardType.name)
+                    cardName = cardType.name,
+                    cardId = cardType.cardBrandId,
+                    isChecked = isChecked
+                ) { checked ->
+                    isChecked = checked
+                    merchantModel = if (isChecked) {
+                        merchantModel.copy(cardTypes = merchantModel.cardTypes + CardBrandType.values().first { it.cardBrandId == cardType.cardBrandId })
```
Comment (timjuic) [inline app/src/main/java/com/Found404/paypro/ui/pages/CardPaymentsPage.kt]: I see that the hardcoded `CardBrandType` is still used here. Card brands received from the backend should only be used.  It is also not clear to me what this line does exactly

File: core/src/main/java/com/found404/core/models/CreditCardType.kt [hunk 2]
```diff
@@ -1,6 +1,13 @@
 package com.found404.core.models
-
+enum class CardBrandType(val cardBrandId: Int, val brandName: String) {
+    DINERS(1, "Diners"),
+    MASTERCARD(2, "MasterCard"),
+    VISA(3, "VISA"),
+    DISCOVER(4, "Discover"),
+    MAESTRO(5, "Maestro");
+}
```
Comment (timjuic) [inline core/src/main/java/com/found404/core/models/CreditCardType.kt]: This should be removed and cardbrands from the backend need to be used. It is enough to define CreditCardType here.

**Review by timjuic — CHANGES_REQUESTED** [review]
Implementation is okay, but some parts of code are not clear what they do. CardBrandType enum needs to be removed as this is exactly what backend returns. So please do that

**Review by jivancic20 — APPROVED** [review]
All good, but make sure to refactor code whenever needed

**Review by timjuic — APPROVED** [review]

**Review by aantonic20 — APPROVED** [review]
All necessary corrections have already been identified and addressed. Remember to refactor the code once everything is completed.

**Pull request merged**